src/kv.hpp

```diff
@@ -0,0 +1,16 @@
+#include <optional>
+#include <string>
+
+using Key = std::string;
+using Value = std::string;
+
+template <typename Store>
+class KV {
+  Store store;
+
+ public:
+  KV<Store>() noexcept;
```

Comment:
May I ask the syntax for this `KV<store>`?

Reply:
Since we are not able to come up with a working implementation of the KV class, we can have different people work on different components. Say, you are dealing with IO and don't care about the internal implementation of the KV class. To make sure you are not blocked, we can mock the interface using some naive substitute like a plain old std::unordered_map. But, another person still needs to implement this interface with a dedicated data structure.

We need this class to be able to have different kinds of implementation. One way is to pass different implementations as a type parameter, which is pretty common in STL - the Allocator type parameter in containers like std::vector.

This way of coding copies that idea. Then, you can have KV<Hashmap>, KV<Skiplist>, ...

Introduction to the syntax can be found here https://en.cppreference.com/w/cpp/language/templates

Reply:
Understood. But for the template class, do we have `<store>` in the constructor? Sry, first time to see this.

Reply:
good call, that's not actually required.

Reply:
Removed in the next commit